src/components/structures/TimelinePanel.js

```diff
@@ -476,9 +476,10 @@ var TimelinePanel = React.createClass({
         // if we are scrolled to the bottom, do a quick-reset of our unreadNotificationCount
         // to avoid having to wait from the remote echo from the homeserver.
         if (this.isAtEndOfLiveTimeline()) {
-            this.props.timelineSet.room.setUnreadNotificationCount('total', 0);
-            this.props.timelineSet.room.setUnreadNotificationCount('highlight', 0);
-            // XXX: i'm a bit surprised we don't have to emit an event or dispatch to get this picked up
```

Comment:
What I would do here is keep the `setUnreadNotificationCount` here and just forceUpdate when the dispatch is caught, otherwise you risk `setUnreadNotificationCount` happening twice if many things want to handle the dispatch.